### PR TITLE
Implement device management enhance

### PR DESCRIPTION
**Pull request by luogangyi**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
/kind feature

**What this PR does / why we need it**:
Device management is a key feature required for IoT use-cases in edge computing.
This proposal addresses how can we enhance device management including more flexible
protocol support, more reasonable device model and device instance design, and
additional way to handle data which is collected from device.

**Which issue(s) this PR fixes**:

Fixes #1687 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
1.  Device model and device instance CRDs changes include below
* upgrade v1alpha1 to v1alpha2
* Move property visitors from device model to device instance.
* Add collectCycle and reportCycle under property visitor.
* Add data section besides twin section.
* Add customized protocol definition in device model and device instance CRDs.
* Add common part in protocol config section
* Extract common part of Modbus protocol config into an independent common part.
* Allow add any customized K-V in protocol config and property visitor section.

2. API changes include below
* device model api no longer need property visitors section
* device api add property visitors section
* device api add report cycle and collect cycle in property visitors section
* device api add report cycle and collect cycle in property visitors section
* device api add customized K-V config in property visitors section
* device api add customized protocol definition section in property visitors section
* device api add customized protocol config section in protocol section
* device api move common part of modbus  protocol into  an independent common part
* device api add CommType, ReconnTimeout, ReconnRetryTimes, CollectTimeout, CollectRetryTimes, CollectType in common part of  protocol section
* device api add customized K-V config in common part of  protocol section
```

### CODE REVIEW

**Comment by luogangyi** [comment]
* Upgrade device api from v1alpha1 to v1alpha2
* Add common part in protocol config section
* Extract common part of Modbus protocol config into an independent common part.
* Allow add any customized K-V in protocol config and property visitor section.

File: cloud/pkg/apis/devices/v1alpha2/device_model_types.go [hunk 1]
```diff
@@ -80,17 +78,6 @@ const (
...
-type DevicePropertyVisitor struct {
-	// Required: The device property name to be accessed. This should refer to one of the
-	// device properties defined in the device model.
-	PropertyName string `json:"propertyName,omitempty"`
-	// Required: Protocol relevant config details about the how to access the device property.
-	VisitorConfig `json:",inline"`
-}
-
 // At least one of its members must be specified.
 type VisitorConfig struct {
```
Comment (fisherxu) [inline cloud/pkg/apis/devices/v1alpha2/device_model_types.go:82]: Move it to device instance :)

File: docs/setup/kubeedge_install_source.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
+# Setup from Source Code
```
Comment (fisherxu) [inline docs/setup/kubeedge_install_source.md:1]: This file should has been deleted or renamed,  so  no need keep it here.

File: docs/setup/kubeedge_install_keadm.md [hunk 3]
```diff
@@ -0,0 +1,374 @@
+# Setup from KubeEdge Installer
```
Comment (fisherxu) [inline docs/setup/kubeedge_install_keadm.md:1]: ditto

File: docs/getting-started/release_package.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
+# Setup using Release package
```
Comment (fisherxu) [inline docs/getting-started/release_package.md:1]: ditto

File: cloud/pkg/apis/devices/v1alpha2/device_model_types.go [hunk 5]
```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2019 The KubeEdge Authors.
```
[the remark this answers is not in the document]
Comment (luogangyi) [reply cloud/pkg/apis/devices/v1alpha2/device_model_types.go]: done :)

**Pull request merged**